### PR TITLE
Filter recommended posts by comom tags

### PR DESCRIPTION
**Pull request by Yndira-FlowForge**
## Description

Added a related filter to recommend articles based on common tags.

Based on https://github.com/11ty/eleventy/discussions/2534#discussioncomment-3415878

## Related Issue(s)

https://github.com/FlowFuse/website/pull/1522#issuecomment-1896299923

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: .eleventy.js [hunk 1]
```diff
@@ -290,6 +290,13 @@ module.exports = function(eleventyConfig) {
         return teamMembers
     });
 
+    eleventyConfig.addNunjucksFilter("related", function (collection = []) {
+        const { tags: requiredTags, page } = this.ctx;
```
Comment (ZJvandeWeg) [inline .eleventy.js:294]: 😮 I didn't know about `this.ctx`. That's great!

**Comment by ZJvandeWeg** [comment]
Great work Yndira! Quick turnaround!

File: src/_includes/layouts/post.njk [hunk 2]
```diff
@@ -65,18 +65,19 @@
                     {% endfor %}
                     <p>Published on: <time value="{{ date | dateToRfc3339 }}">{{ date | shortDate }}</time>
                     </p>
-                    <h3 class="mb-3 pt-6 border-t-2">Recommended Articles:</h3>
-                    <ul class="ml-6 list-disc">
-                        {% for post in collections.posts | reverse | limit(5) %}
-                            {% if post.data.title != title %}
+                    {%- set relatedPosts = collections.posts | related | reverse | limit(5) -%}
```
Comment (ZJvandeWeg) [inline src/_includes/layouts/post.njk:68]:
```suggestion
                    {%- set relatedPosts = collections.posts | relatedPosts | reverse | limit(5) -%}
```

**Comment by ZJvandeWeg** [comment]
cc @Hasmin-AC

**Comment by Yndira-FlowForge** [comment]
Some articles will need to have their tags updated to match and be presented as related, as described in the handbook.

**Pull request merged**